### PR TITLE
Fix update user role issue

### PR DESCRIPTION
**Pull request by zkhin**
Fixes: https://github.com/SeedCompany/cord-api-v3/issues/1221

### CODE REVIEW

File: src/components/user/user.service.ts [hunk 1]
```diff
@@ -402,15 +402,17 @@ export class UserService {
...
+        await this.db
+          .query()
+          .match([
+            node('user', ['User', 'BaseNode'], {
+              id: input.id,
+            }),
+          ])
+          .create([...this.roleProperties(newRoles)])
+          .run();
+      }
```
Comment (CarsonF) [inline src/components/user/user.service.ts:415]: What about removing existing roles not present in the input?

Comment (zkhin) [reply src/components/user/user.service.ts:415]: Okay, I will remove old roles.

File: src/components/user/user.service.ts [hunk 2]
```diff
@@ -401,15 +400,27 @@ export class UserService {
...
           node('user', ['User', 'BaseNode'], {
             id: input.id,
           }),
+          relation('out', 'oldRoleRel', 'roles'),
+          node('oldRoles', 'Property'),
         ])
-        .create([...this.roleProperties(newRoles)])
+        .delete('oldRoleRel')
+        .delete('oldRoles')
+        .run();
```
Comment (CarsonF) [inline src/components/user/user.service.ts:414]: @michaelmarshall you still want active=false here right?

Comment (michaelmarshall) [reply src/components/user/user.service.ts:414]: yes `active` still exists on relationships

Comment (michaelmarshall) [reply src/components/user/user.service.ts:414]: so don't `delete` relationships, just mark them as `active = false`

**Review by michaelmarshall — CHANGES_REQUESTED** [review]

**Review by michaelmarshall — DISMISSED** [review]

**Pull request merged**